06-Preemptive/context_switch.S

```diff
@@ -22,14 +22,10 @@ activate:
 	mrs ip, psr
 	push {r4, r5, r6, r7, r8, r9, r10, r11, ip, lr}
 
-	/* switch to process stack */
-	msr psp, r0
-	mov r0, #3
-	msr control, r0
-	isb
-
 	/* load user state */
-	pop {r4, r5, r6, r7, r8, r9, r10, r11, lr}
+	ldmia r0!, {r4, r5, r6, r7, r8, r9, r10, r11, lr}
```

Comment:
This patch was not intended to fix unexpected exception return behavior. Instead, it eliminates annoying checking that would trigger exception return or not and manipulate saving register on PSP directly.